### PR TITLE
Fix issue querying validator by index for block production

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/chain/evm/node_inquirer.py [hunk 1]
```diff
@@ -732,7 +732,7 @@ def get_transaction_receipt(
             call_order=call_order if call_order is not None else self.default_call_order(),
             tx_hash=tx_hash,
         )
-        assert tx_receipt, 'tx receipt should exist'
+        assert tx_receipt, f'tx receipt should exist {tx_hash.hex()}'
```
Comment (nebolax) [inline rotkehlchen/chain/evm/node_inquirer.py:735]:
```suggestion
        assert tx_receipt, f'tx receipt should exist for tx hash {tx_hash.hex()}'
```

File: rotkehlchen/tasks/manager.py [hunk 2]
```diff
@@ -601,6 +601,7 @@ def _maybe_query_produced_blocks(self) -> Optional[list[gevent.Greenlet]]:
             indices = cursor.execute('SELECT validator_index FROM eth2_validators').fetchall()
```
Comment (nebolax) [inline rotkehlchen/tasks/manager.py:601]: Same comment about unnecessary `fetchall`

File: rotkehlchen/tests/api/test_eth2.py [hunk 3]
```diff
@@ -784,14 +799,14 @@ def test_query_combined_mev_reward_and_block_production_events(rotkehlchen_api_s
...
-    assert len(result['entries']) == result['entries_found'] == result['entries_total'] == 4
+    assert len(result['entries']) == result['entries_found'] == result['entries_total'] == 7
     event_identifier = None
     for entry in result['entries']:
         if entry['entry']['block_number'] == block_number:
             assert entry['grouped_events_num'] == 3
             event_identifier = entry['entry']['event_identifier']
         else:
-            assert entry['grouped_events_num'] == 2
+            assert entry['grouped_events_num'] in (1, 2, 3)
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/tests/api/test_eth2.py]: I've made an explicit check

File: rotkehlchen/tests/api/test_eth2.py [hunk 4]
```diff
@@ -759,6 +764,16 @@ def test_query_combined_mev_reward_and_block_production_events(rotkehlchen_api_s
             'eth2validatorsresource',
         ), json={'validator_index': vindex1},
     )
+    response = requests.put(
+        url=api_url_for(
+            rotkehlchen_api_server,
+            'eth2validatorsresource',
+        ), json={'validator_index': vindex2},
+    )
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_eth2.py:772]: the way you moved the code you are not checking neither this nor the previous API call for correctness, while it was being checked before

```suggestion
    )
    assert_simple_ok_response(response)
```

File: rotkehlchen/tests/api/test_eth2.py [hunk 5]
```diff
@@ -759,6 +764,16 @@ def test_query_combined_mev_reward_and_block_production_events(rotkehlchen_api_s
...
+
+    with rotki.data.db.conn.read_ctx() as cursor:
+        assert cursor.execute('SELECT COUNT(*) FROM eth2_validators').fetchone() == (2,)
+
     assert_simple_ok_response(response)
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_eth2.py:777]:
```suggestion
```

**Review by LefterisJP — APPROVED** [review]
There is some things to fix in the test, but can be merged. Please @yabirgb address them in a subsequent PR. Do not forget

File: rotkehlchen/tests/api/test_eth2.py (hunk 4 above)
Comment (yabirgb) [reply rotkehlchen/tests/api/test_eth2.py:772]: I was only checking the second but there is also a check to assert that the expected number of validators are in the db

**Pull request merged**